### PR TITLE
skip TestDockerFilesystemStats for devicemapper

### LINKED ISSUE #1135: TestDockerFilesystemStats fails on RHEL7 with HasFilesystem:false
HasFilesystem is expected to be true in the test

```
--- FAIL: TestDockerFilesystemStats (10.97s)
        Location:       docker_test.go:336
	Error:		Expected not to be nil.
	Messages:	got info: {Spec:{CreationTime:2016-02-27 03:58:05.94095891 +0000 UTC Aliases:[sharp_aryabhata 75c534be25bd074d7fdf9ae1f88698e8b389cf3fab7392c65023c1ea37f72b33] Namespace:docker Labels:map[] Envs:map[] HasCpu:true Cpu:{Limit:1024 MaxLimit:0 Mask:0-3 Quota:0 Period:0} HasMemory:true Memory:{Limit:18446744073709551615 Reservation:0 SwapLimit:18446744073709551615} HasCustomMetrics:false CustomMetrics:[] HasNetwork:true HasFilesystem:false HasDiskIo:true Image:busybox} Stats:[0xc2080561e0]}
		
FAIL
```

### PR DESCRIPTION
**Pull request by sjenning**
fixes #1135

### CODE REVIEW

File: container/docker/handler.go [hunk 1]
```diff
@@ -168,7 +168,8 @@ func newDockerContainerHandler(
 		ignoreMetrics:      ignoreMetrics,
 	}
 
-	if !ignoreMetrics.Has(container.DiskUsageMetrics) {
+	if !ignoreMetrics.Has(container.DiskUsageMetrics) && storageDriver != devicemapperStorageDriver {
```
Comment (vishh) [inline container/docker/handler.go]: Is this necessary? `otherStorageDir` directory is common to all storage drivers and we do want to surface usage from that.

Comment (sjenning) [inline container/docker/handler.go]: You are saying that we should only not gather stats for the rootfsStorageDir?

Comment (vishh) [inline container/docker/handler.go]: Yeah. `rootfsStorageDir` is storage driver dependent.

**Comment by sjenning** [comment]
oops, didn't mean to close. update pushed to just skip the TestDockerFilesystemStats if devicemapper is the storage driver.

File: integration/tests/api/docker_test.go [hunk 2]
```diff
@@ -295,6 +295,12 @@ func TestDockerFilesystemStats(t *testing.T) {
 	fm := framework.New(t)
 	defer fm.Cleanup()
 
+	storageDriver := fm.Docker().StorageDriver()
+	if storageDriver == framework.DeviceMapper {
```
Comment (vishh) [inline integration/tests/api/docker_test.go]: Logs usage is still supported with devicemapper. If you look below, only the root filesystem usage part is skipped for devicemapper.

**Comment by vishh** [comment]
If we are ok with RHEL not being tested for filesystem stats, I'm fine with this change. We should come back and fix devicemapper support though since it is necessary for proper kube integration.

**Comment by sjenning** [comment]
I think it is fine.

The problem is that ```info.Stats[0].Filesystem``` is nil (```container.Spec.HasFilesystem``` is false) so the test doesn't even get to the ```needsBaseUsageCheck``` part.

**Comment by vishh** [comment]
Should we target ensure that the spec is valid? It should contain logs usage...

**Comment by vishh** [comment]
ok to test

**Comment by vishh** [comment]
Let's merge this for now. But I'd really appreciate any help with turning on the test for RHEL sometime soon. Disk accounting in kube v1.2 will remain untested on RHEL otherwise.

**Comment by vishh** [comment]
@k8s-bot test this

**Pull request merged**